app/org/sagebionetworks/bridge/play/controllers/BaseController.java

```diff
@@ -160,6 +165,22 @@ private String getSessionToken() {
         }
         return session[0];
     }
+    
+    void verifySupportedVersionOrThrowException(Study study, UserSession session) throws UnsupportedVersionException {
+
+    	// Exit early if Admin. These accounts should not throw the exception b/c of version support
+    	// b/c might want to look at older data. The supported version is intended for client apps that
+    	// gather data rather than apps used to model it. syoung 11/19/2015
+    	if (session.isAdminRole()) {
+    		return;  
+    	}
+    	
+        ClientInfo clientInfo = getClientInfoFromUserAgentHeader();
+        String osName = clientInfo.getOsName();
+        if (osName != null && !clientInfo.isSupportedVersion(study.getMinSupportedAppVersion(osName))) {
```

Comment:
null check unneeded. study.getMinSupportedAppVersion() will return null, null is checked for in isSupportedVersion. Which is good because if the osName is unknown, that method will return null (no map entry) as well.